### ISSUE TITLE
Expected a default case.

### ISSUE THREAD

**Issue by jonathanong** [feross/standard#20]
ex:

```js
switch (typeof thing) {
  case 'function':
    return doSomething()
  case 'string':
    return doSomethingElse()
}

throw new TypeError('wtf did u just give me')
```

and i'd make those cases one-liners if i know none of the cases are going to be multiple lines

**Comment by feross** [comment]
So, you want to write `switch` statements without a default case.

I don't feel strongly about this rule. I think fall-through from a `switch` is expected and obvious when none of the cases match. It's just like an `if` statement.

Anyone else have thoughts?

**Comment by dcousens** [comment]
The only advantage with the switch in this case is the implicit LHS of the expression, its like a `with` statement.

IMHO, my list is:

+ Shorter, less repeat of `typeof thing`
+ Default fall-through is obvious, especially with early return
- `break` can be forgotten
- `==` is used, not `===`, therefore breaking one of our standards for quality
- like if, it has 3 possible styles:
```javascript
case "a": return 0
case "b":
  return 1
case "c": {
  return 2
}
```

**Comment by jprichardson** [comment]
I'd be in favor of relaxing this rule.

**Comment by dcousens** [comment]
Right, I was just arguing against a `switch` in general.  Yes the lack of a default case should be allowed.

**Comment by mafintosh** [comment]
my thoughts: `default` should not be required. one-line `case` statements should be allowed (don't know if we already do that).

**Comment by feross** [comment]
Okay :+1: `default` is no longer required in `switch` statements. Released as 2.4.4.

@mafintosh One-line `case` statements are already allowed. :)
